src/tools/advsec.ts

```diff
@@ -0,0 +1,236 @@
+        case "low":
+          return Severity.Low;
+        case "note":
+          return Severity.Note;
+        case "error":
+          return Severity.Error;
+        case "warning":
+          return Severity.Warning;
+        case "undefined":
+          return Severity.Undefined;
+        default:
+          return Severity.Undefined;
+      }
+    });
+  };
+
+  // Helper function to map string confidence levels to enum values
+  const mapConfidenceLevels = (confidenceLevels?: string[]): Confidence[] | undefined => {
+    if (!confidenceLevels) return undefined;
+    return confidenceLevels.map((level) => {
+      switch (level.toLowerCase()) {
+        case "high":
+          return Confidence.High;
+        case "other":
+          return Confidence.Other;
+        default:
+          return Confidence.Other;
+      }
+    });
+  };
+
+  // Helper function to map string validity to enum values
+  const mapValidity = (validity?: string[]): AlertValidityStatus[] | undefined => {
+    if (!validity) return undefined;
+    return validity.map((v) => {
+      switch (v.toLowerCase()) {
+        case "active":
+          return AlertValidityStatus.Active;
+        case "inactive":
+          return AlertValidityStatus.Inactive;
+        case "none":
+          return AlertValidityStatus.None;
+        case "unknown":
+          return AlertValidityStatus.Unknown;
+        default:
+          return AlertValidityStatus.Unknown;
+      }
+    });
+  };
+
+  server.tool(
+    ADVSEC_TOOLS.get_alerts,
+    "Retrieve Advanced Security alerts for a repository.",
+    {
+      project: z.string().describe("The name or ID of the Azure DevOps project."),
+      repository: z.string().describe("The name or ID of the repository to get alerts for."),
+      alertType: z.enum(["code", "dependency", "secret", "unknown"]).optional().describe("Filter alerts by type. If not specified, returns all alert types."),
+      states: z
+        .array(z.enum(["active", "dismissed", "fixed", "autoDismissed", "unknown"]))
+        .optional()
+        .describe("Filter alerts by state. If not specified, returns alerts in any state."),
+      severities: z
+        .array(z.enum(["critical", "high", "medium", "low", "note", "error", "warning", "undefined"]))
+        .optional()
+        .describe("Filter alerts by severity level. If not specified, returns alerts at any severity."),
+      ruleId: z.string().optional().describe("Filter alerts by rule ID."),
+      ruleName: z.string().optional().describe("Filter alerts by rule name."),
+      toolName: z.string().optional().describe("Filter alerts by tool name."),
+      ref: z.string().optional().describe("Filter alerts by git reference (branch). If not provided and onlyDefaultBranch is true, only includes alerts from default branch."),
+      onlyDefaultBranch: z.boolean().optional().default(true).describe("If true, only return alerts found on the default branch. Defaults to true."),
+      confidenceLevels: z
```

Comment:
Can we also add that by default, we fetch both? This has been a point of annoyance from some customers that we only return high confidence alerts.

Reply:
I noticed that the bug applies to every filter criterion that allows multiple values. For instance, if I want to reason on fixed alerts, the query parameter for state is getting set to `criteria.states.0=4` (`4` here is fixed) instead of `criteria.states=4`. Let me reach out to `azure-devops-node-api` Team to figure out if there is anything we can do to mitigate this.

Reply:
Opened microsoft/azure-devops-node-api#644 to fix the bug.